code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -428,7 +438,8 @@
 		if(holder?.drone_controller.drone == src) //if click behaviour has highlighted this drone for control
 			holder.drone_controller.cast(src) //deselect it
 			F.targeting_ability = null
-		src.take_control(usr)
+		if (!isdead(src) && !src.controller && !src.selected_by)
```

Comment:
the second two conditions are already checked in `take_control` and the first probably should be

Reply:
This was to prevent a small case of message spam I think, double clicking a drone without these shows the "This drone is receiving a command!" three times, including once from the first click